web/docs/data-model/operations/queries.md

```diff
@@ -162,20 +162,64 @@ export const getFilteredTasks: GetFilteredTasks<
 }
 ```
 
+<SuperjsonNote />
+
+#### Type support for Queries
+
 Wasp automatically generates the types `GetTasks` and `GetFilteredTasks` based on your Wasp file's declarations:
 
 - `GetTasks` is a generic type automatically generated by Wasp, based on the Query declaration for `getTasks`.
 - `GetFilteredTasks` is also a generic type automatically generated by Wasp, based on the Query declaration for `getFilteredTasks`.
 
-You can utilize these types to define the input and output types for your Query.
+Use these types to type the Query's implementation.
+Annotating Queries is optional, but very helpful.
+Doing so properly types the Query's context.
 
-For example, the Query `getTasks` doesn't expect any arguments (its input type is `void`), but it does return a list of tasks (its output type is `Task[]`).
+In this case, TypeScript will know the `context.entities` object must include the `Task` entity.
+TypeScript also knows whether the `context` object includes user information (it depends on whether your Query uses auth).
 
-On the other hand, the Query `getFilteredTasks` expects an object of type `{ isDone: boolean }`. This type is derived from the `Task` type.
+The generated types are generic and accept two optional type arguments: `Input` and `Output`.
 
-While annotating the Queries is optional, it's highly recommended. Doing so enables **full-stack type safety**. We'll explore what this means when we discuss calling the Query from the client.
+1. `Input` - The argument (i.e., payload) received by the Query function.
+2. `Output` - The Query function's return type.
+
+Use these type arguments to type the Query's inputs and outputs.
+
+For example, the above code says that the Query `getTasks` doesn't expect any arguments (its input type is `void`), but it does return a list of tasks (its output type is `Task[]`).
+On the other hand, the Query `getFilteredTasks` expects an object of type `{ isDone: boolean }`. This type is derived from the `Task` entity type.
+
+Suppose you don't care about typing the Query's inputs and outputs. In that case, you can omit both type arguments, and TypeScript will infer the most general types (i.e., `never` for the input, `unknown` for the output.).
+
+Even though specifying `Input` or `Output` is completely optional, we highly recommend you spcify them. Doing so gives you:
+  - Type support for the arguments and the return value inside the implementation.
+  - **Full-stack type safety**. We'll explore what this means when we discuss calling the Query from the client.
+
+Read more about type support for implementing Queries in the [API Reference](#implementing-queries).
+
+:::tip Inferring the return type
```

Comment:
It's annoying that I have to repeat some of these admonitions twice because they slightly differ (it's usually only mentions of Queries or Actions).

Still, duplication seemed better than coming up with a component that interpolates props into the text.